packages/react-components/lib/robots/utils.ts

```diff
@@ -29,3 +29,53 @@ export function robotModeToString(robotMode: RmfModels.RobotMode): string {
       return `Unknown (${robotMode.mode})`;
   }
 }
+
+export interface VerboseRobot {
+  name: string;
+  model: string;
+  task_id: string;
+  seq: number;
+  mode: RmfModels.RobotMode;
+  battery_percent: number;
+  location: RmfModels.Location;
+  path: RmfModels.Location[];
+  assigned_tasks: RmfModels.TaskSummary[];
+}
+
+export function makeVerboseRobot(
+  robot: RmfModels.RobotState,
+  assignedTasks: RmfModels.TaskSummary[],
+): VerboseRobot {
+  const { name, model, task_id, seq, mode, battery_percent, location, path } = robot;
+  return {
+    name,
+    model,
+    task_id,
+    seq,
+    mode,
+    battery_percent,
+    location,
+    path,
+    assigned_tasks: assignedTasks,
+  };
+}
+
+export function allocateTasksToRobots(
+  robots: RmfModels.RobotState[],
+  tasks: RmfModels.TaskSummary[],
+): VerboseRobot[] {
+  const removableTaskStates = [
+    RmfModels.TaskSummary.STATE_ACTIVE,
+    RmfModels.TaskSummary.STATE_PENDING,
+    RmfModels.TaskSummary.STATE_QUEUED,
+  ];
+  const robotsWithAssignedTasks = robots.map((robot) => {
+    const assignedTasks = tasks.filter((task) => {
```

Comment:
I think we should add a TODO to move this to the server. The server will have a robots endpoints returning the equivalent of `VerboseRobot`.